### PR TITLE
fix(GRANITE-32187) : Massive performance drop in complex dialog openings with coral spectrum

### PR DESCRIPTION
**Pull request by Pareesh**
## Description
While trying to display a complex dialog, there is seen a performance drop. The scripting part of the dialog itself takes around 3000ms( ~3sec).

The related changes done in this PR :
a) Minimise the triggering of event, hence we are only changing the value and triggering event if the new value is different from old /previous stored value.
b) Sometimes callback running in animationFrame gets executed twice/thrice, even if executing them only once is sufficient. Hence avoid running them excessively.

Future Prospects :
a) Forced reflow occurring at some part of code. Try to find and reduce them.
b) If possible try to reduce the animationFrame callbacks . 
c) Reduce duplicated code to reduce downloading and scripting time.

The connectedCallback of the element can be executed more than once as mentioned here : 
https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element-conformance

## Related Issue
https://jira.corp.adobe.com/browse/GRANITE-32187

## Motivation and Context
There has been a performance gap between coralui3 and coral-spectrum and the same code structure takes around 3-4 times longer time to load with spectrum 

## Review
@icaraps @indra2gurjar @CezCz 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: coral-component-accordion/src/scripts/AccordionItem.js [hunk 1]
```diff
@@ -99,7 +99,15 @@ class AccordionItem extends BaseComponent(HTMLElement) {
     return this._selected || false;
   }
   set selected(value) {
-    this._selected = this.hasAttribute('disabled') ? false : transform.booleanAttr(value);
+    let _value = transform.booleanAttr(value);
+    let _selected = this.hasAttribute('disabled') ? false : _value;
+
+    if(this._selected === _selected) {
+      // do nothing if value not changed
```
Comment (CezCz) [inline coral-component-accordion/src/scripts/AccordionItem.js:106]: This comment is redundant. I think we can safely remove it from everywhere and not lose any information.

File: coral-component-masonry/src/scripts/MasonryItem.js [hunk 2]
```diff
@@ -153,10 +153,14 @@ class MasonryItem extends BaseComponent(HTMLElement) {
     if (!this.isConnected) {
       return;
     }
+    // true if component connectedCallback has already been called
+    let isConnected = this._disconnected === false;
```
Comment (CezCz) [inline coral-component-masonry/src/scripts/MasonryItem.js:157]: how about extracting this to a function, it's duplicated in a few places

**Review by CezCz — COMMENTED** [review]
Looks good, however please consider how to fight with code duplication and increasing entropy by removing redundant comments. I consider this a workaround (reflectAttribute should have the check itself, heaviness probably comes from the polyfill) and we'd like to remove it at some point. Will safe up time if is in one place. You may want to leverage Proxy or decorators.

**Comment by Pareesh** [comment]
Hello @CezCz ,
Thanks for the review. I correct the changes according to review comments.

>>> https://github.com/adobe/coral-spectrum/pull/145#pullrequestreview-551306639
I am still checking other places to improves performance. I am targeting small changes in one PR to avoid any break. Thank you for this suggestion. Will try to incorporate this in future PR. 

Thank you

**Review by CezCz — APPROVED** [review]
Proactively approving.

I know that @grubyak is working on Web Components V1, maybe he should take a look at this as well.

**Comment by Pareesh** [comment]
Hello @CezCz ,
I have resolved the conflicts. Please merge the changes, if everything looks good.
Thank you

**Pull request merged**